src/cmd/cmd.zig

```diff
@@ -639,10 +638,16 @@ fn validator() !void {
         geyser_writer,
     );
 
-    // leader schedule
-    var leader_schedule = try getLeaderScheduleFromCli(allocator) orelse
-        try leaderScheduleFromBank(allocator, snapshot.bank.bank_fields);
-    const leader_provider = leader_schedule.provider();
+    // leader schedule cache
+    var leader_schedule_cache = LeaderScheduleCache.init(allocator, snapshot.bank.bank_fields.epoch_schedule);
+    if (try getLeaderScheduleFromCli(allocator) orelse null) |leader_schedule| {
+        try leader_schedule_cache.insertLeaderSchedule(snapshot.bank.bank_fields.epoch, leader_schedule);
+    } else {
+        _ = try leader_schedule_cache.getSlotLeaderMaybeCompute(snapshot.bank.bank_fields.slot, snapshot.bank.bank_fields);
```

Comment:
the double usage of this method is a little confusing - maybe create a helper function `insertLeaderScheduleFromBank` which calls `LeaderSchedule.fromBank(` so its more clear

Reply:
d094aea